spec/tree_spec.rb

```diff
@@ -1,14 +1,110 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(described_class.is_a? Class).to eq true
   end
 end
 
-describe 'Fruit' do
+describe AppleTree do
+  it 'should be a Class' do
+    expect(described_class.is_a? Class).to eq true
+  end
+
+  it 'should set color' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    expect(appleTree.height).to eq 1
+  end
+
+  it 'should set age' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    expect(appleTree.age).to eq 1
+  end
+
+  it 'should set apples' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    expect(appleTree.apples.count).to eq 2
+  end
+
+  it 'should set alive' do
+    apple = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    expect(apple.alive).to eq true
+  end
+
+  it 'age! adds one' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    appleTree.age!
+    expect(appleTree.age).to eq 2
+  end
+
+  it 'age! adds height' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    appleTree.age!
+    expect(appleTree.age).to be > 1
+  end
+
+  it 'any_apples returns true if greater than zero' do
+    appleTree = AppleTree.new(1, 1, [Apple.new("Red", 2.5), Apple.new("Red", 3)], true)
+    expect(appleTree.any_apples?).to eq true
+  end
+
+  it 'any_apples returns false if nil' do
+    appleTree = AppleTree.new(1, 1, nil, true)
+    expect(appleTree.any_apples?).to eq false
+  end
+
+  it 'any_apples returns false if zero' do
+    appleTree = AppleTree.new(1, 1, [], true)
+    expect(appleTree.any_apples?).to eq false
+  end
+
+  it 'dead? returns true if greater than 50' do
+    appleTree = AppleTree.new(1, 55, [], true)
+    expect(appleTree.dead?).to eq true
```

Comment:
rspec also has dynamic predicate matchers which are pretty cool.

https://relishapp.com/rspec/rspec-expectations/v/3-6/docs/built-in-matchers/predicate-matchers